src/d/semantic/vrp.d

```diff
@@ -221,6 +221,13 @@ public:
 			case PtrToInt:
 				assert(0, "Not implemented");
 
+			case SignedToFloat, UnsignedToFloat:
+			case FloatExtend, FloatTrunc:
+				assert(0, "VRP not valid on float-yielding casts");
+
+			case FloatToSigned, FloatToUnsigned:
+				return getRange(e.type.builtin);
+
 				// return ValueRange.get(e.type.builtin);
```

Comment:
That comment belong in the PtrToInt. In fact, PtrToInt should likely do the same as float to int conversion.

Also, all of this needs tests. vrp is especially well tested, and I'd liek to keep it this way.

Reply:
Moved the comment